index.html

```diff
@@ -387,6 +371,49 @@ <h2>
         applied.
       </aside>
     </section>
+    <section>
+      <h2>
+        Calculating an accessible name
+      </h2>
+      <p>
+        The steps for <dfn data-lt="calculate an accessible name">calculating an
+        accessible name</dfn> are given by the following algorithm. The
+        algorithm takes an [=image resource=] |input| and an optional
+        [=Response=] |image| which is the result of [=fetching an image
+        resource=]. It returns a [=string=].
+      </p>
+      <ol>
+        <li>If |input|["label"] is a [=string=]:
+          <ol>
+            <li>Return |input|["label"].
+            </li>
+          </ol>
+        </li>
+        <li>Otherwise if an [=external label=] can be substituted:
+          <ol>
+            <li>Return the relevant [=external label=].
+            </li>
+          </ol>
+        </li>
+        <li>Otherwise if |image| provides an [=embedded accessible name=]:
+          <ol>
+            <li>Return the [=embedded accessible name=].
+            </li>
+          </ol>
+        </li>
+        <li>Otherwise:
+          <ol>
+            <li>Return an empty [=string=] ("").
+            </li>
+          </ol>
+        </li>
+      </ol>
```

Comment:
```suggestion
        <li>If |resource|["label"] is a [=string=], return |input|["label"].</li>
        <li>Otherwise, if an [=external label=] can be substituted, return the relevant [=external label=].</li>
        <li>Otherwise, if |image| provides an [=embedded accessible name=], return the [=embedded accessible name=].</li>
         <li>Return "".</li>
      </ol>
```